dev/archery/requirements-test.txt

```diff
@@ -0,0 +1,2 @@
+pytest
```

Comment:
Could you add the license header?

Reply:
Well, apparently requirements files are excluded from the RAT checks?
I presume comments are not supported in those files.

Reply:
> Well, apparently requirements files are excluded from the RAT checks?

In general, I think that we should use the RAT checks as much as possible for safety. 

> I presume comments are not supported in those files.

https://github.com/apache/arrow/blob/main/docs/requirements.txt uses comment. So we may be able to use comment here too. If we can't, I'm OK with excluding from the RAT checks.

We can defer this to another PR if needed.
I'll merge this.